### PR TITLE
wire-desktop: use system electron, cleanup

### PR DESCRIPTION
**Pull request by max-baz**
Open questions (false positives?):

- [x] wire-desktop E: LICENSES.chromium.html is not a common license (it's not in /usr/share/licenses/common/)
- [ ] wire-desktop E: Dependency nodejs detected and not included (programs ['node'] needed in scripts ['usr/lib/wire-desktop/node_modules/mkdirp/bin/cmd.js', 'usr/lib/wire-desktop/node_modules/loose-envify/cli.js', 'usr/lib/wire-desktop/node_modules/nan/tools/1to2.js', 'usr/lib/wire-desktop/node_modules/sshpk/bin/sshpk-conv', 'usr/lib/wire-desktop/node_modules/ajv/scripts/info', 'usr/lib/wire-desktop/node_modules/sshpk/bin/sshpk-sign', 'usr/lib/wire-desktop/node_modules/uuid/bin/uuid', 'usr/lib/wire-desktop/node_modules/sshpk/bin/sshpk-verify', 'usr/lib/wire-desktop/node_modules/async/support/sync-package-managers.js'])
- [x] wire-desktop W: Dependency included and not needed ('alsa-lib')
- [x] wire-desktop W: Dependency included and not needed ('electron')
- [x] wire-desktop W: Dependency included and not needed ('gconf')
- [x] wire-desktop W: Dependency included and not needed ('gtk2')
- [x] wire-desktop W: Dependency included and not needed ('libxss')
- [x] wire-desktop W: Dependency included and not needed ('libxtst')
- [x] wire-desktop W: Dependency included and not needed ('nss')
- [x] wire-desktop W: Dependency included and not needed ('xdg-utils')

### CODE REVIEW

File: wire-desktop/PKGBUILD [hunk 1]
```diff
@@ -3,25 +3,24 @@
...
 pkgver=3.5.2881
-pkgrel=1
-pkgdesc='Modern, private messenger. Based on Electron.'
-arch=('x86_64' 'i686')
+pkgrel=2
+pkgdesc='End-to-end encrypted messenger with file sharing, voice calls and video conferences'
+arch=('x86_64')
 url='https://wire.com/'
 license=('GPL3')
-conflicts=('wire-desktop-bin')
```
Comment (maximbaz) [inline wire-desktop/PKGBUILD]: `-bin` should conflict with `wire-desktop`, not the other way around.

File: wire-desktop/PKGBUILD [hunk 2]
```diff
@@ -3,25 +3,24 @@
...
-depends=('alsa-lib' 'gconf' 'gtk2' 'libxss' 'libxtst' 'nss' 'xdg-utils')
+depends=('alsa-lib' 'electron' 'gconf' 'gtk2' 'libxss' 'libxtst' 'nss' 'xdg-utils')
 makedepends=('cargo' 'npm' 'python2' 'git' 'yarn')
-optdepends=('hunspell-en: for English spellcheck support'
-            'noto-fonts-emoji: for colorful emoji made by Google'
-            'ttf-emojione: for colorful emoji made by EmojiOne')
-provides=('wire-desktop')
```
Comment (maximbaz) [inline wire-desktop/PKGBUILD]: It's useless to provide the same package name

File: wire-desktop/PKGBUILD [hunk 3]
```diff
@@ -3,25 +3,24 @@
...
+source=("${pkgname}-${pkgver}.tar.gz::https://github.com/wireapp/${pkgname}/archive/linux/$pkgver.tar.gz"
+        "${pkgname}"
         "${pkgname}.desktop")
 sha256sums=('4216cd9c3a2c4920aec2f3c967181b04bfafdb1b47e526a8e823911cce704da1'
+            '13f0829f0995269e7bdf5db683d048d36ae8af8cf16030a581e6386d83aae03a'
             'cc9056cecff2aa49a9ce9c8376d57ec8c7c2cb8174f7966b5cdccbeb2e3751ea')
 
 build() {
-  cd "${srcdir}/${pkgname}-linux-${pkgver}"
+  cd "${pkgname}-linux-${pkgver}"
```
Comment (maximbaz) [inline wire-desktop/PKGBUILD]: `${srcdir}` is the current dir in every function anyway, can save some space by not writing it every time 🙂

File: wire-desktop/PKGBUILD [hunk 4]
```diff
@@ -3,25 +3,24 @@
...
   yarn
   yarn build:ts
   npx grunt 'linux-other'
```
Comment (maximbaz) [inline wire-desktop/PKGBUILD]: This task besides compiling a big binary with bundled electron actually prepares the `wrap` directory. We run this task, but then only take the contents of `wrap/dist/linux-unpacked/resources/app/`, essentially ignoring the binary with bundled electron.

File: wire-desktop/PKGBUILD [hunk 5]
```diff
@@ -30,27 +29,25 @@ build() {
...
 
   # Spellcheck dictionaries
-  rm -rf "${pkgdir}/usr/lib/${pkgname}/resources/app/node_modules/spellchecker/vendor/hunspell_dictionaries"
-  ln -s "/usr/share/hunspell" "${pkgdir}/usr/lib/${pkgname}/resources/app/node_modules/spellchecker/vendor/hunspell_dictionaries"
+  rm -rf "${pkgdir}/usr/lib/${pkgname}/node_modules/spellchecker/vendor/hunspell_dictionaries"
+  ln -s "/usr/share/hunspell" "${pkgdir}/usr/lib/${pkgname}/node_modules/spellchecker/vendor/hunspell_dictionaries"
 
   # Place license files
+  local license
   for license in "LICENSE.electron.txt" "LICENSES.chromium.html"; do
```
Comment (maximbaz) [inline wire-desktop/PKGBUILD]: Without specifying `local`, the loop variable was overriding `license` in the top of the file 😱

File: wire-desktop/PKGBUILD [hunk 6]
```diff
@@ -3,25 +3,24 @@
...
-pkgdesc='Modern, private messenger. Based on Electron.'
-arch=('x86_64' 'i686')
+pkgrel=2
+pkgdesc='End-to-end encrypted messenger with file sharing, voice calls and video conferences'
+arch=('x86_64')
 url='https://wire.com/'
 license=('GPL3')
-conflicts=('wire-desktop-bin')
-depends=('alsa-lib' 'gconf' 'gtk2' 'libxss' 'libxtst' 'nss' 'xdg-utils')
+depends=('alsa-lib' 'electron' 'gconf' 'gtk2' 'libxss' 'libxtst' 'nss' 'xdg-utils')
```
Comment (ConorIA) [inline wire-desktop/PKGBUILD]: It seems electron pulls in alsa-lib, gtk3, libxss, libxtst, and nss. I don't know if gconf and xdg-utils are needed anymore. I'm currently building with electron as the only dep and I'll see if I can get it to run under docker.

File: wire-desktop/PKGBUILD [hunk 7]
```diff
@@ -3,25 +3,24 @@
...
 makedepends=('cargo' 'npm' 'python2' 'git' 'yarn')
```
Comment (ConorIA) [inline wire-desktop/PKGBUILD]: Also not sure if we still need python 2.

File: wire-desktop/PKGBUILD (hunk 6 above)
Comment (ConorIA) [reply wire-desktop/PKGBUILD]: xdg-utils is an opt-dep of electron, and I think it is useful for Wire.

Comment (maximbaz) [reply wire-desktop/PKGBUILD]: Can you tell me what will not work without xdg-utils?

Comment (maximbaz) [reply wire-desktop/PKGBUILD]: I removed `gconf` locally, Wire seems to work fine 🤷‍♂️ Updating the list of dependencies to `electron` and `xdg-utils` alone.

**Pull request merged**